File: BaekJoon/Problems/Step/07/5622.cpp
/* Sample input & output data
WA
13

UNUCIC
36
*/


#include <iostream>
#include <string>

using namespace std;
#define endl "\n"

int main()
{
    // Input
    string s;
    cin >> s;

    // Operate
    int sLength = s.size();
    int dial, time, sum = 0;
    for (int i = 0; i < sLength; i++)
    {
        if (s[i] <= 'R')
        {
            dial = (s[i] - 'A') / 3 + 2;
        } else if (s[i] >= 'S' && s[i] <= 'X')
        {
            dial = (s[i] - 'A' - 1) / 3 + 2;
        } else
        {
            dial = (s[i] - 'A' - 2) / 3 + 2;
        }

        time = dial + 1;
        sum += time;
        cout << s[i] << " " << dial << " " << time << " " << sum << endl;   // test
    }

    // Output
    cout << sum << endl;

    return 0;
}

File: BaekJoon/Problems/Step/06/5622.go
package main
import "fmt"

func main() {

    // Input string s
    var s string
    fmt.Scanln(&s)
    var sRune []rune = []rune(s)

    // Operate
    var sLen int = len(s)
    var dial, time, sum int = 0, 0, 0
    for i := 0; i < sLen; i++ {
        if sRune[i] <= 'R' {
            dial = int(sRune[i] - 'A') / 3 + 2
        } else if sRune[i] >= 'S' && sRune[i] <= 'X' {
            dial = int(sRune[i] - 'A' - 1) / 3 + 2
        } else {
            dial = int(sRune[i] - 'A' - 2) / 3 + 2
        }

        time = dial + 1
        sum += time
    }

    // Output
    fmt.Println(sum)
}